### PR TITLE
Feature/cors lan

### PR DESCRIPTION
**Pull request by kurt-apple**
For the time being, our plan is to offer self-hosting for tdlapp. We should decide where and when to allow certain connections to the API by configuring CORS. As of now this branch adds rules to allow localhost, 127.0.0.1, and 192.168.*.* on all ports to production, however I could foresee creating another environment, TEST, in order for production to really just be the bare minimum of allowed connections.
One thing I'm not sure about is how a highly restrictive CORS allow-list will affect native builds, i.e. to android, or PWAs if that is still a concern.

### CODE REVIEW

File: config/initializers/cors.rb [hunk 1]
```diff
@@ -8,7 +8,10 @@
 
 Rails.application.config.middleware.insert_before 0, Rack::Cors do
   allow do
-    origins Rails.application.config.allowed_cors_origins
+    origins Rails.application.config.allowed_cors_origins,
+      /\Ahttp:\/\/192\.168\.\d+\.\d+(:\d+)?\z/,
+      /\Ahttp:\/\/127\.0\.0\.1(:\d+)?\z/,
+      /\Ahttp:\/\/localhost(:\d+)?\z/
```
Comment (joshbuker) [inline config/initializers/cors.rb]: Rather than appending this to the initializer, you should update the allowed_cors_origins config values under config.

**Review by joshbuker — CHANGES_REQUESTED** [review]
